Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -337,6 +353,101 @@ IPCReply NetKDRequestDevice::HandleNWC24DownloadNowEx(const IOCtlRequest& reques
   return IPCReply(IPC_SUCCESS);
 }
 
+IPCReply NetKDRequestDevice::HandleRequestRegisterUserId(const IOS::HLE::IOCtlRequest& request)
+{
+  auto& system = GetSystem();
+  auto& memory = system.GetMemory();
+
+  INFO_LOG_FMT(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_REQUEST_REGISTER_USER_ID");
+  // Always 0 for some reason, never modified anywhere else
+  memory.Write_U32(0, request.buffer_out + 4);
+
+  if (config.IsRegistered())
+  {
+    WriteReturnValue(NWC24::WC24_ERR_ID_REGISTERED, request.buffer_out);
+    LogError(ErrorType::Account, NWC24::WC24_ERR_ID_REGISTERED);
+    return IPCReply{IPC_SUCCESS};
+  }
+
+  // We require the user's product code.
+  std::string product_code{};
+  const std::string settings_file_path =
+      Common::GetTitleDataPath(Titles::SYSTEM_MENU) + "/" WII_SETTING;
+  const auto fs = m_ios.GetFS();
+  if (const auto file = fs->OpenFile(PID_KD, PID_KD, settings_file_path, FS::Mode::Read))
```

Comment:
can you put the failure path first, so that we can return early on failure and avoid deeply nested levels of indentation?